src/org/sosy_lab/java_smt/test/InterpolatingProverTest.java

```diff
@@ -266,17 +281,41 @@ public <T> void binaryBVInterpolation1() throws SolverException, InterruptedExce
   }
 
   private void requireTreeItp() {
+    requireInterpolation();
+    if (interpolationStrategy == null) {
+      assume()
+          .withMessage("Solver does not support tree-interpolation.")
```

Comment:
You could create a `requireTreeInterpolation()` method that simply checks if an exception is thrown, similarly as for sequential interpolants or interpolation in general.
This goes for all occurrences of this in this class.